### PR TITLE
Coin meta

### PR DESCRIPTION
**Pull request by johnnyluo**

### CODE REVIEW

File: VultisigApp/VultisigApp/Views/Components/Cells/ChainSelectionCell.swift [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import SwiftUI
 
 struct ChainSelectionCell: View {
-    let assets: [Coin]
+    let assets: [CoinMeta]
```
Comment (flypaper0) [inline VultisigApp/VultisigApp/Views/Components/Cells/ChainSelectionCell.swift:11]: nice

File: VultisigApp/VultisigApp/Chains/AccountFactory.swift [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+import Foundation
+import WalletCore
+
+struct CoinFactory {
+
+    private init() { }
+
+    static func create(asset: CoinMeta, hexPubKey: String, hexChainCode: String) throws -> Coin {
+        let derivePubKey = PublicKeyHelper.getDerivedPubKey(
```
Comment (johnnyluo) [inline VultisigApp/VultisigApp/Chains/AccountFactory.swift:16]: For EdDSA chain, public key doesn't need to derive

File: VultisigApp/VultisigApp/View Models/CoinSelectionViewModel.swift [hunk 3]
```diff
@@ -236,18 +120,17 @@ class CoinSelectionViewModel: ObservableObject {
     
     
     private func addToChain(asset: CoinMeta, to vault: Vault, priceProviderId: String?) async throws -> Coin? {
-        guard let newCoin = getNewCoin(asset: asset, vault: vault) else {
-            return nil
-        }
+        let newCoin = try CoinFactory.create(
+            asset: asset,
+            hexPubKey: vault.pubKeyECDSA,
```
Comment (johnnyluo) [inline VultisigApp/VultisigApp/View Models/CoinSelectionViewModel.swift:125]: here , solana , SUI and polkadot is using EdDSA key , not ECDSA

File: VultisigApp/VultisigApp/Views/Vault/CustomTokenView.swift [hunk 4]
```diff
@@ -165,35 +156,22 @@ struct CustomTokenView: View {
...
-                        self.tokenDecimals = decimals
-                        self.showTokenInfo = true
-                        self.isLoading = false
-                        
-                    }
+                    self.tokenName = name
+                    self.tokenSymbol = symbol
+                    self.tokenDecimals = decimals
+                    self.showTokenInfo = true
+                    self.isLoading = false
```
Comment (enriquesouza) [inline VultisigApp/VultisigApp/Views/Vault/CustomTokenView.swift:174]: This will change the way Custom Tokens works now, is that what we want?

**Review by flypaper0 — APPROVED** [review]
lgtm

**Pull request merged**